fix: FIT-639: Remove strman library dependency and replace with lodash

luarmr opened the pull request:
# Remove strman library and replace with lodash

## 🎯 Purpose
Remove the strman library dependency and replace its usage with lodash functions and a minimal custom utility.

## 🔧 Changes Made

### Dependencies Removed
- `strman` package from all package.json files
- `strman.tostudlycaps` package  
- `@types/strman` type definitions
- Updated webpack externals configuration

### Function Replacements
| strman Function | Replacement |
|----------------|-------------|
| `toCamelCase` | `camelCase` (lodash) |
| `toSnakeCase` | `snakeCase` (lodash) |
| `toKebabCase` | `kebabCase` (lodash) |
| `toStudlyCaps` | Custom utility (PascalCase not in lodash) |


## 🧪 Testing
- [x] Linter passes with no new errors
- [x] All existing functionality preserved
- [x] Yarn install successful
- [x] String conversion functions work identically

## 📊 Impact
- **Bundle size reduction**: Removed unnecessary dependency
- **Consistency**: All string utilities now use lodash or minimal custom code
- **Maintainability**: Fewer dependencies to manage
- **Architecture**: Clean separation with utilities only in open source

## 🔍 Architecture Notes
- Custom utilities only exist in open source (`@humansignal/core`)
- Enterprise version imports utilities via `@humansignal/core` at build time
- Direct lodash usage preferred over wrapper functions
- Only one custom utility needed (`toStudlyCaps` for PascalCase)


<img width="1903" height="1075" alt="Screenshot 2025-08-29 at 10 25 14 AM" src="https://github.com/user-attachments/assets/32df3a65-68b2-49d7-88d8-65ee577e2523" />
<img width="349" height="413" alt="Screenshot 2025-08-28 at 2 57 15 PM" src="https://github.com/user-attachments/assets/ea121cd6-8b99-4d0e-af45-ff2338030257" />
<img width="346" height="395" alt="Screenshot 2025-08-28 at 2 57 08 PM" src="https://github.com/user-attachments/assets/a609d542-61cb-4d87-b778-5182967207d7" />

luarmr committed:
fix: FIT-639: Remove strman library dependency and replace with lodash

nass600 commented on web/apps/labelstudio/src/pages/CreateProject/Config/Config.jsx:
```diff
@@ -17,7 +17,7 @@ import { TemplatesList } from "./TemplatesList";
 import { UnsavedChanges } from "./UnsavedChanges";
 import { Checkbox, CodeEditor, Select } from "@humansignal/ui";
-import { toSnakeCase } from "strman";
+import { snakeCase } from "lodash";
```
this probably should be `import snakeCase from "lodash/snakeCase"` and this applies to all the rest of changes

luarmr replied:
Hilarious that I made that mistake. Thanks!!!

luarmr replied:
I need to look into biomne plugins to see if we can protect this for happening.

luarmr committed:
fix: Replace destructuring lodash imports with tree-shakable imports

Replace problematic destructuring imports from main lodash package:
- import { camelCase } from 'lodash' ❌
+ import camelCase from 'lodash/camelCase' ✅

Files fixed:
- libs/editor/src/utils/utilities.ts (camelCase, snakeCase)
- apps/labelstudio/src/pages/CreateProject/Config/Config.jsx (snakeCase)
- libs/editor/src/tools/Base.jsx (kebabCase)
- libs/editor/src/LabelStudio.tsx (camelCase)
- libs/datamanager/src/sdk/dm-sdk.js (camelCase)
- libs/datamanager/src/utils/helpers.ts (camelCase)
- libs/core/src/lib/utils/string.ts (camelCase)

This prevents webpack from loading the entire lodash library and enables
proper tree-shaking for optimal bundle size.
